### PR TITLE
[CARBONDATA-2120]Fixed is null filter issue

### PR DESCRIPTION
**Pull request by kumarvishal09**
**Problem**: Is null filter is failing for numeric data type(No dictionary column).

**Root cause:** Min max calculation is wrong when no dictionary column is not the first column. 

As it is not the first column null value can come in between and min max for null value is getting updated only when first row is null

**Solution:** Update the min max in all the case when value is null or not null for all type

  - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
 Added UT      
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2208/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3449/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3313/

**Comment by kumarvishal09** [comment]
retest sdv please

**Comment by ravipesala** [comment]
LGTM

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3329/

File: core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public CarbonTablePath(CarbonTableIdentifier carbonTableIdentifier, String table
    * @param carbonFilePath
    */
   public static String getFolderContainingFile(String carbonFilePath) {
-    return carbonFilePath.substring(0, carbonFilePath.lastIndexOf(File.separator));
+    return carbonFilePath.substring(0, carbonFilePath.lastIndexOf('/'));
```
Comment (scwf) [inline core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java]: why this change?
